docs/pages/pmd/userdocs/extending/testing.md

```diff
@@ -195,41 +197,49 @@ in a "CDATA" section, so that no further XML escapes (entity references such as
             <message>Violation message 2</message>
         </expected-messages>
         <code><![CDATA[
-    public class ConsistentReturn {
-        public Boolean foo() {
-        }
+public class ConsistentReturn {
+    public Boolean foo() {
     }
-         ]]></code>
-            <source-type>apex</source-type>                             <!-- optional -->
-        </test-code>
+}
+        ]]></code>
+        <source-type>apex</source-type>                             <!-- optional -->
```

Comment:
I would also explain why source-type is optional, and not needed in most cases. It's just used to select a specific language version, not the language. Most PRs that add a new language add source-type in the xml files, but this is not needed and throws off schema validation, because of this line:

https://github.com/pmd/pmd/blob/0cc00277d348b8cdbf294d38b8a0268fd92195d9/pmd-test/src/main/resources/rule-tests_1_0_0.xsd#L53

Schema validation fails until the schema is updated.


I'm not sure it's documented somewhere, how to update the schema (eg on which server to put the new file). This is an item in #597

I think anyway, this validation constraint should be removed from the schema. It forces us to update the schema each time